packages/perseus/src/widgets/group/validate-group.ts

```diff
@@ -0,0 +1,31 @@
+import {emptyWidgetsFunctional} from "../../renderer-util";
+
+import type {PerseusStrings} from "../../strings";
+import type {ValidationResult} from "../../types";
+import type {
+    PerseusGroupUserInput,
+    PerseusGroupValidationData,
+} from "../../validation.types";
+
+function validateGroup(
+    userInput: PerseusGroupUserInput,
+    validationData: PerseusGroupValidationData,
+    strings: PerseusStrings,
+    locale: string,
+): ValidationResult {
+    const emptyWidgets = emptyWidgetsFunctional(
```

Comment:
This makes me happy that we're able to reuse the functional helpers like this rather than having to access methods on components.